alpine-server/src/main/java/alpine/server/auth/OidcAuthenticationService.java

```diff
@@ -182,6 +186,7 @@ private OidcUser authenticateInternal(final OidcProfile profile) throws AlpineAu
                     LOGGER.debug("Assigning subject identifier " + profile.getSubject() + " to user " + user.getUsername());
                     user.setSubjectIdentifier(profile.getSubject());
                     user.setEmail(profile.getEmail());
+                    customizer.onAuthenticationSuccess(profile, idToken, accessToken);
                     return qm.updateOidcUser(user);
```

Comment:
Just to be safe, since `updateOidcUser` could still fail:

```suggestion
                    user = qm.updateOidcUser(user);
                    customizer.onAuthenticationSuccess(profile, idToken, accessToken);
                    return user;
```

Same for other invocations. You want to make sure `onAuthenticationSuccess` is only called when you can be sure that the authentication succeeded.

Reply:
Do you think `onAuthenticationSuccess` might benefit from accepting and/or returning an `OidcUser`? So this would become something like

```java
return customizer.onAuthenticationSuccess(qm.updateOidcUser(user), profile, idToken, accessToken);
```

That way, implementers would have access to `user.setTeams()`, `user.setPermissions()`, etc. inside `onAuthenticateSuccess`, if you think that would be useful and if it seems like an acceptable design pattern